catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -240,59 +245,17 @@ protected Object executeWithSubject() throws Exception {
             StandardThreadFactoryBuilder.newThreadFactory("dumpCommandThread"),
             rejectedExecutionHandler);
 
-    while (!response.getResults().isEmpty()) {
-      response =
-          new SourceResponseImpl(
-              new QueryRequestImpl(query, props),
-              ResultIterable.resultIterable(
-                      catalog::query, new QueryRequestImpl(query, props), pageSize)
-                  .stream()
-                  .collect(Collectors.toList()));
-
-      if (StringUtils.isNotBlank(zipFileName)) {
-        try {
-          Optional<QueryResponseTransformer> zipCompressionTransformer = getZipCompression();
-          if (zipCompressionTransformer.isPresent()) {
-            BinaryContent binaryContent =
-                zipCompressionTransformer.get().transform(response, zipArgs);
-            if (binaryContent != null) {
-              IOUtils.closeQuietly(binaryContent.getInputStream());
-            }
-            Long resultSize = (long) response.getResults().size();
-            printStatus(resultCount.addAndGet(resultSize));
-          }
-        } catch (InvalidSyntaxException e) {
-          LOGGER.info("No Zip Transformer found.  Unable export metacards to a zip file.");
-        } catch (CatalogTransformerException e) {
-          LOGGER.info("zipCompression transform failed");
-        }
-      } else if (multithreaded > 1) {
-        final List<Result> results = new ArrayList<>(response.getResults());
-        executorService.submit(
-            () -> {
-              for (final Result result : results) {
-                Metacard metacard = result.getMetacard();
-                try {
-                  exportMetacard(dumpDir, metacard, resultCount);
-                } catch (IOException e) {
-                  LOGGER.debug("Failed to dump metacard {}", metacard.getId(), e);
-                }
-              }
-            });
-      } else {
-        for (final Result result : response.getResults()) {
-          Metacard metacard = result.getMetacard();
-          exportMetacard(dumpDir, metacard, resultCount);
-        }
-      }
-
-      if (response.getResults().size() < pageSize || pageSize == -1) {
-        break;
-      }
-
-      query.setStartIndex(query.getStartIndex() + pageSize);
+    QueryRequest queryRequest = new QueryRequestImpl(query, props);
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Hits for Search: {}", catalog.query(queryRequest).getHits());
     }
 
+    ResultIterable.resultIterable(catalog::query, queryRequest)
+        .stream()
+        .map(Collections::singletonList)
+        .map(result -> new SourceResponseImpl(queryRequest, result))
+        .forEach(response -> handleResult(response, executorService, dumpDir, resultCount));
+
     executorService.shutdown();
 
     while (!executorService.isTerminated()) {
```

Comment:
`catch (InterruptedException)` below should not ignore the exception. It should propagate it by re-interrupting the thread outside of the `while()` in a `finally` clause if the exception was ever received.